### ISSUE TITLE
app support for status-im:// URIs in the browser - windows

### ISSUE THREAD

**Issue by iurimatias** [status-im/status-desktop#837]
i.e going to these urls in the browser should open the status desktop app

e.g go to profile -> my profile -> copy profile url; open in browser and click "Open in Status"

**Comment by jrainville** [comment]
A beginning of this task was done here https://github.com/status-im/nim-status-client/pull/965
but it seems like we'll need to update the Registry on Windows and make a call as root on Linux, which can be a problem for us, since we do not have a real installer. We just unzip the app where needed and run it.

One possibility would be to have button in the app the first time you run it where once pressed, asks your for elevated privileges, once we get those, we can update the registry or equivalent on other OS, then the user can close the app and reopen.

Another issue I faced is that I don't know how to **not** open a new instance of the app when calling the protocol URl. Right now, I can use the URI to open the app and go to the right channel, but if the app is already opened, it will open the app another time, which is not wanted.

Other apps like Discord open the wanted channel in the same app, but I'm not sure how they do it. I looked at QT's documentation and Windows' one and couldn't find a real solution.

One possibility I thought of would be to have a file descriptor next to the app executable that when the app is opened, it starts watching the file descriptor and when a URI is called, it calls a bash script or something that sets the file descriptor to the URI, which procs an event in the already opened app, otherwise we open a new app. (docs for SocketNotifier https://doc.qt.io/qt-5/qsocketnotifier.html#details)

It's probably not the standard way to do it, but like I said, I couldn't find a standard way.

Hopefully, you are someone with experience with desktop app and protocol URIs and can lend a hand 😄

**Comment by iurimatias** [comment]
and join.status.im urls in browser

**Comment by hesterbruikman** [comment]
Any thoughts on supporting https://eips.ethereum.org/EIPS/eip-681 ?
(e.g. ethereum:pay-0x8c4f71b3cf6a76de2cc239a6fa84e1a80e589598?value=5e15)

Assuming this would be another issue altogether, wondering what the added complexity is once status-im:// direct links are supported

**Comment by jrainville** [comment]
> Any thoughts on supporting https://eips.ethereum.org/EIPS/eip-681 ?
> (e.g. ethereum:pay-0x8c4f71b3cf6a76de2cc239a6fa84e1a80e589598?value=5e15)
> 
> Assuming this would be another issue altogether, wondering what the added complexity is once status-im:// direct links are supported

It would be possible with what we know and have implemented already. This EIP is a bit more complex because there was more parameters that can be passed through the url, but nothing too difficult.

Either way, you're right, it would be done through another issue. Also, would we want another design for the transaction window to show that this is from the URL? Doesn't need to be a completely different modal, just add more explanation.

**Comment by hesterbruikman** [comment]
> Either way, you're right, it would be done through another issue. Also, would we want another design for the transaction window to show that this is from the URL? Doesn't need to be a completely different modal, just add more explanation.

Good point, afaik we currently don't distinguish where a request is coming from on mobile. Is that what you mean? Some sort of indicator on the modal, that the request is coming from a uri? I'll create a design issue for that

I'll create a separate issue in status-desktop for support of eip-681. Don't think it's urgent, but good for any wallet to support wallet standards. Looked a bit into Bitpay and Coinpayments. Both seem to push their own API, not sure why that is

**Comment by jrainville** [comment]
> Is that what you mean? Some sort of indicator on the modal, that the request is coming from a uri? I'll create a design issue for that

Yes exactly. Doesn't need to be complex, just something so the user understand what he clicked.

> 'll create a separate issue in status-desktop for support of eip-681. Don't think it's urgent, but good for any wallet to support wallet standards.

I agree 👍

**Comment by iurimatias** [comment]
blocked by #2223 so icing this until that is done. a .bat solution is available on PR #2043

**Comment by iurimatias** [comment]
let's start with windows first, then do this on mac
